src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -1221,24 +1280,24 @@ def setup_trunk_build(bucket_paths, fuzz_target, build_prefix=None):
 def setup_regular_build(revision,
                         bucket_path=None,
                         build_prefix='',
-                        fuzz_target=None) -> RegularBuild:
+                        fuzz_target=None) -> Optional[RegularBuild]:
   """Sets up build with a particular revision."""
   if not bucket_path:
     # Bucket path can be customized, otherwise get it from the default env var.
     bucket_path = get_bucket_path('RELEASE_BUILD_BUCKET_PATH')
 
-  build_urls = get_build_urls_list(bucket_path)
   job_type = environment.get_value('JOB_NAME')
-  if not build_urls:
-    logs.error('Error getting build urls for job %s.' % job_type)
-    return None
-  build_url = revisions.find_build_url(bucket_path, build_urls, revision)
-  if not build_url:
-    logs.error(
-        'Error getting build url for job %s (r%d).' % (job_type, revision))
+  build_url = _get_build_url(bucket_path, revision, job_type)
 
+  if not build_url:
     return None
 
+  all_bucket_paths = _get_build_bucket_paths()
```

Comment:
nit: rather than recompute this stuff here, can we just pass a `is_latest` argument to `setup_regular_build` (which defaults to false)?

Reply:
It is possible that a latest build gets fetched without going through the setup_trunk_build codepath, so I don't think that would work.

Reply:
Yes, it is possible, but I don't think we need to care about them? 

We really only care about fuzz_task here (which calls setup_trunk_build). Yes, there are some cases where another task will happen to setup the latest build, but I don't think we need to track those?

Reply:
The design doc states that some non fuzz taks are relevant, so maybe these non-trunk cases are relevant. @alhijazi any thoughts on this?

Reply:
For tasks like progression, we care about running the latest build as well to be able to tell if an issue was fixed or not. This impacts the time between a fix being introduced and clusterfuzz marking an issue as fixed.
Same for analyze, we would like to know if a given issue reproduces at head.